rotkehlchen/tests/unit/test_protocol_balances.py

```diff
@@ -356,6 +359,47 @@ def test_gmx_balances(
     }
 
 
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('arbitrum_one_accounts', [['0x3D3db31AdbCCB6a4e5aE94ec0701d361E147FA6A']])
+@pytest.mark.parametrize('arbitrum_one_manager_connect_at_start', [(get_arbitrum_allthatnode(weight=ONE, owned=True),)])  # noqa: E501
+@pytest.mark.parametrize('should_mock_current_price_queries', [False])
+def test_gmx_balances_staking(
+        arbitrum_one_inquirer: 'ArbitrumOneInquirer',
+        arbitrum_one_transaction_decoder: 'OptimismTransactionDecoder',
+        arbitrum_one_accounts: list[ChecksumEvmAddress],
+        arbitrum_one_manager_connect_at_start,
+        inquirer: 'Inquirer',  # pylint: disable=unused-argument
+) -> None:
+    """
+    Test querying balances for GMX. We use an address with 2 different positions and the other
+    one has a single position. There are both shorts and longs in the data.
+    """
+    wait_until_all_nodes_connected(
+        connect_at_start=arbitrum_one_manager_connect_at_start,
+        evm_inquirer=arbitrum_one_inquirer,
+    )
+    get_decoded_events_of_transaction(
+        evm_inquirer=arbitrum_one_inquirer,
+        database=arbitrum_one_transaction_decoder.database,
+        tx_hash=deserialize_evm_tx_hash('0x25160bf17e5a77935c7661933c045739dba44606859a20f00f187ef291e56a8f'),
+    )
+    balances_inquirer = GmxBalances(
+        database=arbitrum_one_transaction_decoder.database,
+        evm_inquirer=arbitrum_one_inquirer,
+    )
+
+    # we mock the function call for the user positions since the function returns a set and
+    # the inconsistency in the order can affect the output of the mocked responses. We do
+    # check that the expected output and the actual output match.
+    balances = balances_inquirer.query_balances()
```

Comment:
where do you mock the function? I don't see it.

Also ... what about the other account/position?